demisto_sdk/commands/pre_commit/pre_commit_command.py

```diff
@@ -347,19 +356,39 @@ def group_by_language(
             infra_files.append(file)
 
     language_to_files: Dict[str, Set] = defaultdict(set)
-    integrations_scripts = []
+    integrations_scripts: Set[IntegrationScript] = set()
     for integration_script_paths in more_itertools.chunked_even(
         integrations_scripts_mapping.keys(), INTEGRATIONS_BATCH
     ):
         with multiprocessing.Pool(processes=cpu_count()) as pool:
-            integrations_scripts.extend(
-                pool.map(BaseContent.from_path, integration_script_paths)
-            )
+            chunk = pool.map(BaseContent.from_path, integration_script_paths)
+            for i in chunk:
+                if not i or not isinstance(i, IntegrationScript):
+                    continue
+                integrations_scripts.add(i)
     exclude_integration_script = set()
     for integration_script in integrations_scripts:
-        if not integration_script or not isinstance(
-            integration_script, IntegrationScript
-        ):
+        if (pack := integration_script.in_pack) and pack.object_id == API_MODULES_PACK:
+            # add api modules to the api_modules list, we will handle them later
+            api_modules.append(integration_script)
+            continue
+    if api_modules:
+        update_content_graph(graph)
+        api_modules: List[Script] = graph.search(
+            object_id=[api_module.object_id for api_module in api_modules]
+        )  # type: ignore[no-redef]
+        for api_module in api_modules:
+            for imported_by in api_module.imported_by:
+                # we need to add the api module for each integration that uses it, so it will execute the api module check
+                assert isinstance(imported_by, Integration)
```

Comment:
```suggestion
               if not isinstance(imported_by, Integration):
                   raise RunTimeError("....")
```
or

```suggestion
                assert isinstance(imported_by, Integration), f'add-error....'
```

Would be more informative then just an assert

Reply:
this is for type checking, i changed the return value for `imported_by` function as it has to be Script no matter what

Reply:
but won't it crash the code if type isn't as expected? do we want it to crash?